File: src/main/handcraft-react/src/Models/Requests/CheckoutRequest.ts
import CheckoutBill from "../CheckoutBill";

class CheckoutRequest {
  firstName: string;
  lastName: string;
  street: string;
  apartment: string;
  city: string;
  zip: string;
  email: string;
  phone:string;
  checkoutBill: CheckoutBill;
  constructor(
    firstName: string,
    lastName: string,
    street: string,
    apartment: string,
    city: string,
    zipCode: string,
    email: string,
    phone:string,
    checkoutBill: CheckoutBill
  ) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.street = street;
    this.apartment = apartment;
    this.city = city;
    this.zip = zipCode;
    this.email = email;
    this.phone = phone;
    this.checkoutBill = checkoutBill;
  }
}
export default CheckoutRequest;

File: src/main/java/com/ecommerce/handcraft/requests/CheckoutRequest.java
package com.ecommerce.handcraft.requests;

import com.ecommerce.handcraft.entity.CheckoutBill;
import lombok.Data;

@Data
public class CheckoutRequest {
    private String firstName;
    private String lastName;
    private String street;
    private String apartment;
    private String city;
    private String zip;
    private String email;
    private String phone;
    private CheckoutBill checkoutBill;
}
